### DISCUSSION TITLE
Support for PartNet-Mobility Dataset

### DISCUSSION THREAD

**Discussion by beneisner** [pyg-team/pytorch_geometric#2793] [Ideas]
(Similar in theme to #2791, but I created a separate issue so they could be discussed separately)

The PartNet-Mobility Dataset is a 3D dataset that provides a bunch of fine-grained semantic and instance labels for a large set of **articulated** 3D objects. The dataset contains both meshes and point clouds. **Additionally, it contains articulations for each object, which describe how individual parts move in relation to one another**, i.e. what direction a drawer slides. It is an extension of the ShapeNet effort, which this pytorch_geometric already supports.

I already have an implementation of this dataset that I'm using in my research, and would be happy to upstream it. Would such a contribution be welcome?

**Comment by rusty1s** [comment]
It's also very much welcome, although we should probably start with one of the datasets.

**Comment by beneisner** [comment]
@rusty1s I'm getting around to this now (hopefully). This particular dataset is somewhat complicated and has several different aspects to it; do you tend to require that the entire dataset object fit into a single file in datasets/ ? Or can it span multiple files (i.e. a utils file, etc.)

**Reply by rusty1s** [reply]
I think it can span multiple files. If you think this is the way to go, please feel free to propose a first version :)
